libmamba/src/core/link.cpp

```diff
@@ -978,8 +974,7 @@ namespace mamba
             throw std::runtime_error(std::string("Path type not implemented: ")
                                      + std::to_string(static_cast<int>(path_data.path_type)));
         }
-        // TODO we could also use the SHA256 sum of the paths json
-        return std::make_tuple(validate::sha256sum(dst), rel_dst);
+        return std::make_tuple(path_data.sha256, rel_dst);
```

Comment:
I think we should change this line to check if `path_data.sha256` has a value. Old packages might not have shipped a `paths.json` file and thus there might not be a value available.

Something like `path_data.sha256.empty() ? validate::sha256sum(dst) : path_data.sha256` should do the trick.

Reply:
good catch! would it also make sense to change this to an optional then in the PathData to make this explicit? Or is that a bit overkill?

Reply:
Yeah we currently treat strings as optional by abusing the empty property.
I think we could start using optional more, but maybe we should do that more consistently in more places.

Reply:
Cool! I changed it! :)